src/Interpreters/Aggregator.cpp

```diff
@@ -2631,45 +2685,69 @@ void NO_INLINE Aggregator::mergeStreamsImplCase(
     {
         /// Merge state of aggregate functions.
         aggregate_functions[j]->mergeBatch(
-            0, rows,
+            row_begin, row_end,
             places.get(), offsets_of_aggregate_states[j],
-            aggregate_columns[j]->data(),
+            aggregate_columns_data[j]->data(),
             aggregates_pool);
     }
-
-    /// Early release memory.
-    block.clear();
 }
 
 template <typename Method, typename Table>
 void NO_INLINE Aggregator::mergeStreamsImpl(
-    Block & block,
+    Block && block,
```

Comment:
This will not work as expect. It's still pass-by-reference (r-value) and it will not be destructed after function ends.

Reply:
Indeed, thanks, fixed.